builtin/providers/google/resource_google_project.go

```diff
@@ -298,3 +376,65 @@ func mergeBindings(bindings []*cloudresourcemanager.Binding) []*cloudresourceman
 
 	return rb
 }
+
+func resourceServices(d *schema.ResourceData) []string {
+	// Calculate the tags
+	var services []string
+	if s := d.Get("services"); s != nil {
+		ss := s.(*schema.Set)
+		services = make([]string, ss.Len())
+		for i, v := range ss.List() {
+			services[i] = v.(string)
+		}
+	}
+	return services
+}
+
+func resourceServicesDelta(d *schema.ResourceData) (added, removed []string) {
+	// Calculate the tags
+	if d.HasChange("services") {
+		oldS, newS := d.GetChange("services")
+		oldSs := oldS.(*schema.Set)
+		newSs := newS.(*schema.Set)
+
+		oldMap := make(map[string]bool)
+		newMap := make(map[string]bool)
+
+		for _, v := range oldSs.List() {
+			oldMap[v.(string)] = true
+		}
+		for _, v := range newSs.List() {
+			newMap[v.(string)] = true
+		}
+
+		added = make([]string, 0)
+		removed = make([]string, 0)
```

Comment:
nit: `var added, removed []string` does effectively the same thing and is a bit clearer, in my opinion.